snuba/query/composite.py

```diff
@@ -62,6 +62,30 @@ def __init__(
         )
         self.__from_clause = from_clause
 
+    def __repr__(self) -> str:
+        from snuba.query.formatters.tracing import format_query
+
+        # NOTE (Vlad): Why the type is ignored:
+        # If you remove the ignore type comment you will get the following error:
+        #
+        #   Argument 1 to "format_query" has incompatible type
+        #   "CompositeQuery[TSimpleDataSource]"; expected
+        #   "Union[Query, CompositeQuery[SimpleDataSource]]"
+        #
```

Comment:
What happens if we make the `format_query` function generic with respect to `TSimpleDataSource`.
`format_query(Union[Query, CompositeQuery[TSimpleDataSource]])`
Since TsimpleDatSource is bound to SimpleDataSource, you should still be able to access the SimpleDataSource methods

Reply:
I tried it, this is what will happen:

```
Argument 1 to "visit" of "TracingQueryFormatter" has 
incompatible type "Union[
    Entity, 
    ProcessableQuery[TSimpleDataSource], 
    CompositeQuery[TSimpleDataSource],
    JoinClause[TSimpleDataSource]
]"; 
expected "Union[
    JoinClause[<nothing>], 
    ProcessableQuery[<nothing>], 
    CompositeQuery[<nothing>]
]
```

Reply:
mh, ok.